File: jimmer-ddd-java-spring-boot-starter/src/test/java/org/morecup/jimmerddd/java/spring/domain/goods/Goods.java
package org.morecup.jimmerddd.java.spring.domain.goods;

import org.babyfish.jimmer.sql.Entity;
import org.morecup.jimmerddd.java.spring.domain.BaseEntity;

@Entity
public interface Goods extends BaseEntity {
    String name();
    String nowAddress();
}

File: Jimmer-ddd-kotlin-spring-boot-starter/src/test/kotlin/org/morecup/jimmerddd/kotlin/spring/domain/goods/Goods.kt
package org.morecup.jimmerddd.kotlin.spring.domain.goods

import org.babyfish.jimmer.sql.Entity
import org.morecup.jimmerddd.kotlin.aggregateproxy.KAggregateProxy
import org.morecup.jimmerddd.kotlin.spring.domain.BaseEntity

@Entity
interface Goods : BaseEntity {
    val name: String
    val nowAddress: String
}
val goodsAggregateProxy = KAggregateProxy(GoodsDraft::class)
class GoodsImpl(goods: GoodsDraft) : GoodsDraft by goods {
    fun rename(newName: String) {
        this.name = newName
    }
}